src/fides/connectors/aws.py

```diff
@@ -96,19 +103,52 @@ def describe_dynamo_tables(client: Any, table_names: List[str]) -> List[Dict]:
     Returns describe_table response given a 'dynamodb' boto3 client.
     """
     describe_tables = []
-    for table in table_names:
-        described_table = client.describe_table(TableName=table)
-        describe_tables.append(described_table["Table"])
+    for table_name in table_names:
+        described_table = describe_dynamo_table(client, table_name)
+        if isinstance(described_table, dict):
+            describe_tables.append(described_table)
+
     return describe_tables
 
+@handle_common_aws_errors
+def describe_dynamo_table(client: Any, table_name: str) -> Dict:  # type: ignore
+    """
+    Returns describe_table response given a 'dynamodb' boto3 client.
+    """
+    described_table = client.describe_table(TableName=table_name)
+    described_table["Table"]["Fields"] = scan_dynamo_table(client, table_name)
+
+    return described_table["Table"]
+
 
 @handle_common_aws_errors
 def get_dynamo_tables(client: Any) -> List[str]:  # type: ignore
     """
     Returns a list of table names response given a 'rds' boto3 client.
     """
     list_tables = client.list_tables()
-    return list_tables["TableNames"]
+    table_names = list_tables["TableNames"]
+    next_page_exists = "LastEvaluatedTableName" in list_tables
+    while next_page_exists:
+        last_evaluated_table_name = list_tables["LastEvaluatedTableName"]
+        list_tables = client.list_tables(ExclusiveStartTableName=last_evaluated_table_name)
+        table_names.extend(list_tables["TableNames"])
+        next_page_exists = "LastEvaluatedTableName" in list_tables
+    return table_names
+
+@handle_common_aws_errors
+def scan_dynamo_table(client: Any, table_name: str) -> List[str]:  # type: ignore
+    """
+    Returns describe_table response given a 'dynamodb' boto3 client.
+    """
+    table_scan = client.scan(TableName=table_name, Limit=30)
+    fields = set()
+    for item in table_scan["Items"]:
+        for field, _ in item.items():
+            fields.add(field)
```

Comment:
could we do something like this to make it a bit more usable in the plus detection code?
```suggestion
def scan_dynamo_table(client: Any, table_name: str, num_samples: int = 30) -> List[str]:  # type: ignore
    """
    Returns describe_table response given a 'dynamodb' boto3 client.
    """
    table_scan = client.scan(TableName=table_name, Limit=num_samples)
    fields = set()
    for item in table_scan["Items"]:
        for field, _ in item.items():
            fields.add(field)
```

Reply:
i'm just gonna commit this, i don't think there's any harm!